### ISSUE TITLE
BenchmarksGame tests fail if not run from dotnet/coreclr build directory

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#14989]
The following tests:
```
JIT\Performance\CodeQuality\BenchmarksGame\k-nucleotide
JIT\Performance\CodeQuality\BenchmarksGame\regex-redux
JIT\Performance\CodeQuality\BenchmarksGame\reverse-complement
```

use some code in `harness-helpers.cs` to find the data file they want to run with. It has a fragile, strong dependence on the precise dotnet/coreclr repro build. If you move the tests w.r.t. the CORE_ROOT, they will fail.

Worse, they fail by throwing an exception that the file is not found, but nobody catches the exception, so it just appears the app crashes.

So:
1. The tests should look, first, without requiring CORE_ROOT to be set, in the current directory.
2. The failure should be a message, not an uncaught exception.

**Comment by AndyAyersMS** [comment]
IIRC we need the exception here to ensure failures get properly reported when running under xunit-perf. 

Since that testing mode does not invoke `Main` I suppose we could put a try-catch in Main to keep the exception from escaping in the EXE case.

For the "find input files issue" we have been waiting for the Netstandard 1.5 update to allow us to query where in the file system the exe resides since in general the input files will be in the same directory. So I'd rather not add more ad-hoc searching logic here if possible...

**Comment by BruceForstall** [comment]
> For the "find input files issue" we have been waiting for the Netstandard 1.5 update to allow us to query where in the file system the exe resides since in general the input files will be in the same directory.

When is that expected to happen? And why not Netstandard 2.0?

**Comment by BruceForstall** [comment]
BTW, none of the BenchmarkGame tests are running on arm32/arm64 in official runs due to https://github.com/dotnet/coreclr/issues/14977.

**Comment by AndyAyersMS** [comment]
Update is tracked by #14124. Not sure when or who is on tap for this.

Historically we would backport the benchmark fixes to the 1.1 release branch, so the aim was to keep the dependency to no higher than netstandard 1.6. Not sure if we care that much about 1.1 backports now, so perhaps netstandard 2.0 would be fine.

**Comment by BruceForstall** [comment]
Going to close this since #14124 covers it
